ext/curl/interface.c

```diff
@@ -1625,6 +1625,7 @@ static int curl_debug(CURL *cp, curl_infotype type, char *buf, size_t buf_len, v
 	if (type == CURLINFO_HEADER_OUT) {
 		if (ch->header.str) {
 			zend_string_release_ex(ch->header.str, 0);
+			ch->header.str = NULL;
```

Comment:
Is the problematic case where buf_len == 0, which means that this releases ch->header.str, but due to the if in the following line, it will not be overwritten by a new buffer?

Reply:
I found this problem in versions 5.5-5.6. The script crashed with the message "zend_mm_heap is corrupted", but support for these versions was already completed, and I did not check new versions.

Then the fall was caused by freeing memory without resetting the string length to zero. As I understand it, the re-request caused an attempt to free memory that was no longer owned by the variable. But I'm not sure since I'm not an expert.

We have now switched to versions 7 and 8 and the same problem occurred. The script crashes steadily on the second request with the message "Bus error".

I will try to write some code to check for this error.

Reply:
I didn't manage to reproduce anything either, but I found a related issue, which is fixed by https://github.com/php/php-src/commit/30e791ed560a4682adfb8188d7bc1486ff086c23.

Reply:
I merged a variant of your fix in https://github.com/php/php-src/commit/8c292a2f9d20151d7269f8b95d5ddde226c8efb8. Rather than nulling out header.str, it drops the empty string check to make sure the old pointer is always overwritten.

Reply:
I applied this patch on 8.0.10 and the problem disappeared completely. Thanks.